lib/services/sites.js

```diff
@@ -74,23 +79,68 @@ function getSites() {
     siteConfig.assetDir = normalizeDirectory(siteConfig.assetDir || 'public');
     siteConfig.prefix = siteConfig.path && siteConfig.path.length > 1 ? siteConfig.host + siteConfig.path : siteConfig.host;
 
-    // check to make sure sites have default properties
-    if (!siteConfig.host) {
-      throw new Error('Missing host in site config');
-    }
-
     if (!siteConfig.protocol) {
       log('warn', `A protocol property ('http' or 'https') was not found for site '${site}', it will default to 'http'`);
       siteConfig.protocol = 'http';
     }
 
-    // add site config to the sites map
-    fullConfig[siteConfig.slug] = siteConfig;
+    return siteConfig;
+  }
+
+  files.getFolders(instanceSitesFolder).forEach(site => {
```

Comment:
**Suggestion**: run a `.map` to create an array with `subsiteDir` per each file. Then run a separate `.foreach` to check if `files.fileExists`. This way we avoid having two nested for loops.

Reply:
I'm not sure there's a good way around having O(n^2) here... I've been messing around with it for a little while. `subsiteDir` has to be looped to get all of the subsites inside of the directory, whether that happens here or in another for loop, it'll still be nested :(